distributed/diagnostics/tests/test_worker_plugin.py

```diff
@@ -54,30 +71,32 @@ async def test_create_on_construction(c, s, a, b):
 
 @gen_cluster(nthreads=[("127.0.0.1", 1)], client=True)
 async def test_normal_task_transitions_called(c, s, w):
-    expected_transitions = [
-        ("task", "waiting", "ready"),
-        ("task", "ready", "executing"),
-        ("task", "executing", "memory"),
+    expected_notifications = [
+        Transition("task", "waiting", "ready"),
+        Transition("task", "ready", "executing"),
+        Transition("task", "executing", "memory"),
+        ReleasedKey("task", "memory"),
     ]
 
-    plugin = MyPlugin(1, expected_transitions=expected_transitions)
+    plugin = MyPlugin(1, expected_notifications=expected_notifications)
 
     await c.register_worker_plugin(plugin)
     await c.submit(lambda x: x, 1, key="task")
+    await asyncio.sleep(DELAY_BEFORE_TEARDOWN)
```

Comment:
I understand the desire to avoid flaky tests, however if the test depends on a sleep, then it may be sensitive to delays.  For example, on small CI machines we frequently experience GC cleanups on the order of seconds.  So if this test isn't robust to random multi-second-long pauses with no work then it is likely to fail eventually.